### DISCUSSION TITLE
Welcome to jPOS Discussions!

### DISCUSSION THREAD

**Discussion by ar** [jpos/jPOS#377] [General]
## 👋 Welcome!

  We’re using Discussions as a place to connect with other members of our community. We hope that you:

  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by vishnupillai** [comment]
Thanks Alejandro for starting this discussion. I'm Vishnu Janardhanan and I am a JPOS user and contributor. I have used JPOS to implement many financial services and contributed a bit when I could. Currently, I'm mostly an observer and might start contributing again when possible.

**Comment by juanbjs** [comment]
Hello, I'm Juan Zarate, I'm Software Developer, I want to implement JPOS, but I want to connect JPOS and a Web Service(REST or SOAP), is this possible?

**Reply by ar** [reply]
Sure, you can certainly do that using jPOS, and possibly its TransactionManager.

**Reply by juanbjs** [reply]
Let me explain you, what I need.

 I have socket transactions, I want to get the trancaction on Socket and send it in a web service...

<img width="659" alt="Screen Shot 2021-03-04 at 18 48 27" src="https://user-images.githubusercontent.com/37712256/110035317-42e72b80-7d1a-11eb-8303-32124085990a.png">

**Reply by ar** [reply]
I don't think the 'Welcome to jPOS Discussions!' thread is the right one for questions like this. You can certainly do what you want, but you need to develop the REST/SOAP code, an probably integrate it as a jPOS participant. Take a look at http://jpos.org/tutorials to get some initial ideas.

**Comment by iidah409** [comment]
Hi All, My name is Idah Idris. I am a software developer also a jpos user. I have used jpos framework to implement a major POS switching. Currently an observer and hoping to contribute when possible.

**Reply by ar** [reply]
Welcome!

**Comment by ronlhayes** [comment]
Greetings! We are a .Net shop and I am trying a POC to see if we can run JPOS as a Docker Container that we call using a REST API. I tried to create an image using the ubuntu_vivid_jdk8 dockerfile but ran into the errors shown in the attached JPG. I am running this in Windows using Docker Desktop. Any ideas? Also, is there additional generic documentation aside from the REST API sample in http://jpos.org/doc/tutorials/jpos-rest.pdf? Is that covered in the paid documentation? Does JPOS expose a swagger API? Thanks in advance for any advice.

![ubuntu_vivid_jdk8 errors](https://user-images.githubusercontent.com/1558257/220405163-9e1bd5a7-7cb1-444d-b0e2-ea7367f6c95d.jpg)

**Reply by ronlhayes** [reply]
Appears the REST API is only included with JPOS EE? Do you have any Docker files for that repo?

**Reply by barspi** [reply]
jPOS-EE is a collection of modules (each published as a maven lib)
It's not meant to be run inside a Docker
Maybe it's better for you to start a new project from the jPOS template  https://github.com/jpos/jPOS-template/
Add the necessary dependencies and code, and build a docker image from it

**Reply by barspi** [reply]
We have successfully used Amazon's Corretto images such as
`FROM amazoncorretto:8` or `FROM amazoncorretto:11`

**Reply by ronlhayes** [reply]
Thanks for the timely useful information! Looking closer to see if a licensed version might make more sense.

**Comment by gregorioosorio** [comment]
Hi,

jPOS EE is available in github: https://github.com/jpos/jPOS-EE and also
you can find maven artifacts here:
https://mvnrepository.com/artifact/org.jpos.ee

Remember to read the license (AGPL) and check if that fits to your business.

Regarding the docker issue: notice that the jpos template uses Java 11. So,
you have to adjust your Dockerfile in order to use the propper version of
Java.

You should be able to follow the rest API tutorial, build a Docker image
and run it  in a Docker container.

Regards,

**Reply by ronlhayes** [reply]
Unfortunately I am not proficient with all the details of using Java so would I simply change all references of Java8 to Java11? I'm in a bit of a quandary. After reading the license agreement we will likely need a commercial license however until I can prove it meets our needs by creating a docker image that I can make REST calls (JPOS with the REST API from EE) against it will be a harder sell to management. Let me ponder and possibly send an email with proposed solution.

**Reply by barspi** [reply]
@ronlhayes 
Are you trying to prove if a jPOS-based system servicing a REST API can be run inside a Docker container, or are you trying to prove that **you can do it**?  Two very different questions :-)

The answer to the first one is a 110% YES.  A large part of the payments world, from little startups, to Big Names, is running on top of some jPOS-based system, with or without APIs, virtualized or on-the-iron, you name it.
I don't think how that can be a hard sell, since jPOS is the de facto industry standard.

As for the second question... well, why don't you tell us what you're trying to do, exactly?
Maybe you can write to info@transactility.com for internal discussion on license plans and support.

**Reply by ronlhayes** [reply]
We have a meeting scheduled with Constantino Voulgaris on 3/6 to discuss.

**Comment by ronlhayes** [comment]
Appears that my colleague Dmitri Boudtchenko has already initiated contact with your sales department. We are trying to coordinate the right team for a call to discuss details including licensing.
